### DISCUSSION TITLE
possibility of supporting c++20 coroutine syntax?

### DISCUSSION THREAD

**Discussion by epsilon-0** [uNetworking/uWebSockets#1515] [Ideas]
I was wondering if it will be possible to support the co_await (and friends) syntax at some point in the future.

**Comment by uNetworkingAB** [comment]
That's probably quite easy to do actually, and you could even do it as a completely separate layer above the callback layer.

All you need is to wrap all the functions that are "async" into being promises and whatnot, and shove all that down a mutable lambda or anything that can be passed to the functions taking lambda.

**Comment by epsilon-0** [comment]
Interesting, would that be on top of usockets or uwebsockets? I don't know how to do it right now, but I could try to see how.

**Comment by uNetworkingAB** [comment]
Coroutines are one layer above callbacks. So it's entirely above uWS.

**Comment by uNetworkingAB** [comment]
You probably want something like CoroutineApp
